pkg/contexts/ocm/transfer/transferhandler/transferhandler.go

```diff
@@ -11,12 +11,65 @@ import (
 	"github.com/open-component-model/ocm/pkg/errors"
 )
 
-type TransferOptions interface{}
+const KIND_TRANSFEROPTION = "transfer option"
 
+// TransferOptions is the general type for an option bag
+// holding options for a transfer handler.
+// Different transfer handler implementations may use differ
+// concrete option sets.
+// Therefore, every option set uses its own options type
+// The options setters/getters MUST be interfaces, which may be implemented
+// by different option set types to enable the option implementations
+// to work with different options sets.
```

Comment:
This makes very little sense for me as a user of the library.

What is an option bag? Why is it a bag? Why would I care about all of these? Why is this introduced right now?

Is there a simpler way to do all of this?

Reply:
I'll add an explanation, but basically a typical user never has to deal with this type. This in only relevant for implementing standard handlers with their own set of options.

To clearly separate those use cases, it would be required to introduce multiple packages and because of the arising dep cycles, again an internal package would be required, which would complicate the package structure even more. Because of this I've put everything, the user relevant parts and the implementation specific parts into a single package,

Reply:
I'll support this separation of concern by making the transfer package self-contained for the typical use case.
So, it will provide all the types required to use it. There is no need anymore to deal with the transferhandler package in the import list.

If the user wants to explicitly use a standard handler he can use the appropriate handler package.
Only if the user wants to provide an custom handler  to implement custom logic, he has to deal with the transferhandler package.